server/asyncio_extensions.py

```diff
@@ -32,6 +46,50 @@ async def gather_without_exceptions(
     return results
 
 
+# Based on python3.8 asyncio.Lock
+# https://github.com/python/cpython/blob/6c6c256df3636ff6f6136820afaefa5a10a3ac33/Lib/asyncio/locks.py#L106
+class SpinLock(_ContextManagerMixin):
```

Comment:
Yea it just calls `acquire` and `release` https://github.com/python/cpython/blob/3.9/Lib/asyncio/locks.py. The other option would be to inherit from `Lock` and override the functions, but that's messy because they share very little implementation.